File: C++/2652. Sum Multiples.cpp
class Solution {
public:
    int cal(int n, int k)
    {
        return k * (n / k) * (n / k + 1) / 2;
    }
    int sumOfMultiples(int n) {
        return cal(n, 3) + cal(n, 5) + cal(n, 7) - cal(n, 3 * 5) - cal(n, 5 * 7) - cal(n, 3 * 7) + cal(n, 3 * 5 * 7);
    }
};

File: Python/2652. Sum Multiples.py
class Solution:
    def cal(self, n, k):
        return k * (n // k) * (n // k + 1) // 2

    def sumOfMultiples(self, n: int) -> int:
        return (
            self.cal(n, 3)
            + self.cal(n, 5)
            + self.cal(n, 7)
            - self.cal(n, 3 * 5)
            - self.cal(n, 5 * 7)
            - self.cal(n, 3 * 7)
            + self.cal(n, 3 * 5 * 7)
        )